pmd-core/src/main/java/net/sourceforge/pmd/renderers/CSVRenderer.java

```diff
@@ -95,7 +95,14 @@ public CSVRenderer(ColumnDescriptor<RuleViolation>[] columns, String theSeparato
     }
 
     public CSVRenderer() {
-        this(ALL_COLUMNS, DEFAULT_SEPARATOR, PMD.EOL);
+        super(NAME, "Comma-separated values tabular format.");
+
+        separator = DEFAULT_SEPARATOR;
+        cr = PMD.EOL;
+
+        for (ColumnDescriptor<RuleViolation> desc : allColumns) {
+            definePropertyDescriptor(booleanPropertyFor(desc.id, desc.title));
+        }
```

Comment:
this seems unrelated to the rest of the PR… what are we fixing here exactly?

Reply:
I needed to change the property  descriptors for the default columns to non-static, so that I can access the instance method `determineFileName` of AbstractRenderer/CSVRenderer.